### PR TITLE
Feature/multi fluentbit support

### PR DESCRIPTION
**Pull request by ahma**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: pkg/sdk/api/v1beta1/node_agent_types.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/banzaicloud/operator-tools/pkg/typeoverride"
+	"github.com/banzaicloud/operator-tools/pkg/types"
+	"github.com/banzaicloud/operator-tools/pkg/volume"
+)
+
+// +kubebuilder:object:generate=true
+
+type NodeAgent struct {
+	Name          string              `json:"name,omitempty"`
+	Type          string              `json:"type,omitempty"`
```
Comment (pepov) [inline pkg/sdk/api/v1beta1/node_agent_types.go:27]: We must emphasize here and in the docs that based on the given type, the operator will automatically configure the required nodeselectors and tolerations for the workload.

We need to highlight the fact that a fully custom configuration is also supported.

File: pkg/resources/nodeagent/daemonset.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+func (n *nodeAgentInstance) daemonSet() (runtime.Object, reconciler.DesiredState, error) {
+	var containerPorts []corev1.ContainerPort
+	if n.nodeAgent.FluentbitSpec.Metrics != nil && n.nodeAgent.FluentbitSpec.Metrics.Port != 0 {
+		containerPorts = append(containerPorts, corev1.ContainerPort{
+			Name:          "monitor",
+			ContainerPort: n.nodeAgent.FluentbitSpec.Metrics.Port,
+			Protocol:      corev1.ProtocolTCP,
+		})
+	}
```
Comment (nickgerace) [inline pkg/resources/nodeagent/daemonset.go:47]: Just a question: these ports are only required for the metrics server?

File: pkg/resources/nodeagent/nodeagent.go [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+				DB:              util.StringPointer("/tail-db/tail-containers-state.db"),
+				MemBufLimit:     "5MB",
+				Tag:             "kubernetes.*",
+			},
+			Security: &v1beta1.Security{
+				RoleBasedAccessControlCreate: util.BoolPointer(true),
+				SecurityContext:              &v1.SecurityContext{},
+				PodSecurityContext:           &v1.PodSecurityContext{},
+			},
+			ContainersPath: "/var/lib/docker/containers",
```
Comment (nickgerace) [inline pkg/resources/nodeagent/nodeagent.go:99]: Is this going to use the `mountPath` field from the `fluentbitSpec` eventually?

File: pkg/resources/nodeagent/nodeagent.go [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+		},
+		ContainersPath: "C:\\ProgramData\\docker",
+		VarLogsPath:    "C:\\var\\log",
+		DaemonSetOverrides: &typeoverride.DaemonSet{
+			Spec: typeoverride.DaemonSetSpec{
+				Template: typeoverride.PodTemplateSpec{
+					Spec: typeoverride.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:    containerName,
```
Comment (nickgerace) [inline pkg/resources/nodeagent/nodeagent.go:217]: I have two questions here.
1. Is this going to be exposed via `values.yaml`?
2. Is the Dockerfile for this going to be included in this PR?
CC: @cbron

**Review by nickgerace — COMMENTED** [review]
Thanks for doing this again! Had a few comments.

File: pkg/resources/nodeagent/daemonset.go (hunk 2 above)
Comment (ahma) [reply pkg/resources/nodeagent/daemonset.go:47]: Yes, and also for the liveness check

File: pkg/resources/nodeagent/nodeagent.go (hunk 4 above)
Comment (pepov) [reply pkg/resources/nodeagent/nodeagent.go:217]: 1. yes
2. yes, we will include it as an example, just won't setup any automated build for it. once you have a build and image in place we can use and add it as the default windows image

Comment (nickgerace) [reply pkg/resources/nodeagent/nodeagent.go:217]: Okay great. I'll get started on that: https://github.com/rancher/fluent-bit-package

File: pkg/resources/nodeagent/daemonset.go (hunk 2 above)
Comment (nickgerace) [reply pkg/resources/nodeagent/daemonset.go:47]: Ok great. Thank you

File: fluent-bit-image/v1.6/Dockerfile.windows [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+ARG WINDOWS_VERSION=ltsc2019
+
+#
+# Builder Image - Windows Server Core
+#
+FROM mcr.microsoft.com/windows/servercore:$WINDOWS_VERSION as builder
+
+# The FLUENTBIT_VERSION ARG must be after the FROM instruction
+ARG FLUENTBIT_VERSION=1.7.0
```
Comment (nickgerace) [inline fluent-bit-image/v1.6/Dockerfile.windows:11]: @ahma should this directory be called `v1.7` since the Dockerfile uses Fluentbit 1.7?

**Review by nickgerace — APPROVED** [review]

File: charts/logging-operator-logging/templates/logging.yaml [hunk 6]
```diff
@@ -54,4 +54,9 @@ spec:
   {{- else }}
   fluentbit: {}
   {{- end }}
-
+{{- if .Values.nodeAgents }}
+{{ toYaml .Values.nodeAgents | indent 4 }}
+    {{- end }}
+{{- else }}
+    nodeAgents: {}
+{{- end }}
```
Comment (nickgerace) [inline charts/logging-operator-logging/templates/logging.yaml:62]: Wouldn't we be missing the `nodeAgents:` field for this? Would it be better to do the following?

```yaml
{{- with .Values.nodeAgents }}
nodeAgents:
  {{- toYaml . | nindent 4 }}
{{- end }}
```

**Review by nickgerace — CHANGES_REQUESTED** [review]

**Comment by nickgerace** [comment]
Seems like there may be a helm chart build error. Hmmm I am unable to see, otherwise I would help :)
https://github.com/banzaicloud/logging-operator/pull/665/checks?check_run_id=2249718349

**Review by tarokkk — APPROVED** [review]

File: charts/logging-operator-logging/templates/logging.yaml [hunk 7]
```diff
@@ -54,4 +54,9 @@ spec:
   {{- else }}
   fluentbit: {}
   {{- end }}
-
+{{- if .Values.nodeAgents }}
+{{ toYaml .Values.nodeAgents | indent 4 }}
+    {{- end }}
```
Comment (tarokkk) [inline charts/logging-operator-logging/templates/logging.yaml:59]:
```suggestion
```

Comment (ahma) [reply charts/logging-operator-logging/templates/logging.yaml]: Thanks!

**Pull request merged**